src/Components/Components/src/Dispatcher.cs

```diff
@@ -10,34 +10,66 @@ namespace Microsoft.AspNetCore.Components
     /// <summary>
     /// Dispatches external actions to be executed on the context of a <see cref="Renderer"/>.
     /// </summary>
-    public interface IDispatcher
+    public abstract class Dispatcher
     {
+        /// <summary>
+        /// Creates a default instance of <see cref="Dispatcher"/>.
+        /// </summary>
+        /// <returns>A <see cref="Dispatcher"/> instance.</returns>
+        public static Dispatcher CreateDefault() => new RendererSynchronizationContextDispatcher();
+
+        /// <summary>
+        /// Provides notifications of unhandled exceptions that occur within the dispatcher.
+        /// </summary>
+        internal event UnhandledExceptionEventHandler UnhandledException;
```

Comment:
This is cool. I like this as a way of making keeping the actual `event` nonpublic (so random things can't subscribe to it) while still being able to raise it from subclasses.